collections/_article/release-candidate-godot-4-2-rc-2.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Release candidate: Godot 4.2 RC 2"
+excerpt: "The first release candidate brought another crop of regression reports, which we've now fixed!"
+categories: ["pre-release"]
+author: Rémi Verschelde
+image: /storage/blog/covers/release-candidate-godot-4-2-rc-2.webp
+image_caption_title: "2D platformer prototype"
+image_caption_description: "An open source game by Securas"
+date: 2023-11-24 17:00:00
+---
+
+Since our first release candidate for Godot 4.2 a week ago, a number of regressions have been reported by new testers who were starting to test using Godot 4.2 in production. So we've been busy this week triaging, testing, debugging, and fixing regressions to reach a point where we're confident that we're ready for the stable release.
+
+That's where this second release candidate comes in, polished as much as we can and ready to be trialed on all your Godot 4.1 projects, hopefully as a smooth upgrade! If you notice anything off that's not already listed under the [known issues](#known-issues) below, please make sure to report it quickly so we can keep track of it in our release planning.
+
+This release candidate includes a number of big changes that warrant particular attention:
+
+- Many users have been running into spurious `slot >= slot_max` errors in their release exports in Godot 4.0 and 4.1, typically around changing scenes or otherwise manipulating nodes in the scene tree. We finally managed to debug and fix it properly ([GH-85280](https://github.com/godotengine/godot/pull/85280)), which should solve a lot of issues, including crashes on Windows. This change should find its way in a 4.1.x maintenance release too.
+- For Windows builds, we kept running into issues with MinGW's implementation of `std::thread` and related threading classes. We replaced it with the third-party [mingw-std-threads](https://github.com/meganz/mingw-std-threads) library, which seems to solve the issues we had, and should also improve performance. Integrating such a library so late in the release cycle is a risky change, but we ran out of alternative to workaround those threading issues, and the first test results are promising. Please make sure to test our Windows builds heavily on projects using multithreading.
+- The tool we made to upgrade 4.1 mesh surfaces to the new 4.2 format works well, but the way it showed itself as an immediate dialog could cause some rendering deadlocks in the editor. So we changed it to a less intrusive warning, which will encourage you to run the tool yourself via the 'Project > Tools > Upgrade Mesh Surfaces..' entry. It's easier to miss, but it's better than a frozen editor.
+- As a reminder, since beta 5, we provide pre-compiled ARM64 and ARM32 Linux binaries (editor and templates). This RC 2 improves this new set of binaries by temporarily disabling <abbr title="Link Time Optimization">LTO</abbr>, which seemed to cause crashes on at least Raspberry Pi OS. Please try it out if you have such ARM Linux systems!
```

Comment:
```suggestion
- As a reminder, since beta 5, we provide pre-compiled ARM64 and ARM32 Linux binaries (editor and templates). RC 2 improves this new set of binaries by temporarily disabling <abbr title="Link Time Optimization">LTO</abbr>, which seemed to cause crashes on at least Raspberry Pi OS. Please try it out if you have such ARM Linux systems!
```